changelog.js

```diff
@@ -0,0 +1,77 @@
+const core = require('@actions/core')
+const gh = require('@actions/github')
+
+const filesToCheck = ['CHANGELOG.md', 'py/CHANGELOG.md']
+
+const context = gh.context
+// We have to have the github token set for fetching PR description and content of the files.
+const oktokit = gh.getOctokit(process.env.GITHUB_TOKEN)
+
+// Get the PR link for current pull request.
+const prLink = `([#${context.payload.pull_request.number}](${context.payload.pull_request.html_url}))`
+
+// Make the title pretty.
+function getCleanTitle (title) {
+  // remove fix(component): prefix
+  title = title.split(': ').slice(-1)[0].trim()
+  // remove links to JIRA tickets, i.e. a suffix like [ISSUE-123]
+  title = title.split('[')[0].trim()
+  // remove trailing dots
+  title = title.replace(/\.+$/, '')
+
+  return title
+}
+
+// Checks if the file contains changelog entry, returns `true` if found or `false` otherwise.
+async function changelogExists (path) {
+  // Get the file content from the pull request branch.
+  const { data } = await oktokit.rest.repos.getContent({ owner: context.repo.owner, repo: context.repo.repo, ref: context.ref, path })
+  const buf = Buffer.alloc(data.content.length, data.content, data.encoding)
+  const fileContent = buf.toString()
+
+  if (core.isDebug()) {
+    core.debug(`Checking file: ${path}.`)
+    core.debug(`${fileContent}`)
+  }
+
+  return fileContent.includes(prLink)
+}
+
+async function main () {
```

Comment:
This is just a nitpick, but if you compare this new function with the old implementation, then it is much harder to read now. Could we get this closer to the old implementation, especially since this tries to do exactly the same as before? There are a few more debug outputs now, but I'm not sure if we really need them.